src/crawlers.js

```diff
@@ -1,105 +1,187 @@
             qs: queryString,
         });
     },
 
-    startCrawler: (requestPromise, options) => {
-        if (!options.crawler) {
-            throw new Error('Missing required parameter: crawler');
-        }
-        if (!options.token) {
-            throw new Error('Missing required parameter: token');
-        }
+    createCrawler: (requestPromise, options) => {
+        const { userId, token, settings } = options;
+
+        checkParamOrThrow(userId, 'userId', 'String');
+        checkParamOrThrow(token, 'token', 'String');
+        checkParamOrThrow(settings, 'settings', 'Object');
+        checkParamOrThrow(settings.customId, 'settings.customId', 'String');
+
+        const requestParams = {
+            json: true,
+            method: 'POST',
+            url: `${options.baseUrl}${BASE_PATH}/${userId}/crawlers`,
+            qs: { token },
+            body: settings,
+        };
+
+        return requestPromise(requestParams);
+    },
+
+    updateCrawler: (requestPromise, options) => {
+        const { userId, token, settings, crawlerId } = options;
+
+        checkParamOrThrow(userId, 'userId', 'String');
+        checkParamOrThrow(token, 'token', 'String');
+        checkParamOrThrow(crawlerId, 'crawlerId', 'String');
+        checkParamOrThrow(settings, 'settings', 'Object');
+
+        const requestParams = {
+            json: true,
+            method: 'PUT',
+            url: `${options.baseUrl}${BASE_PATH}/${userId}/crawlers/${crawlerId}`,
+            qs: { token },
+            body: settings,
+        };
+
+        return requestPromise(requestParams);
+    },
+
+    getCrawlerSettings: (requestPromise, options) => {
+        const { userId, token, crawlerId } = options;
+
+        checkParamOrThrow(userId, 'userId', 'String');
+        checkParamOrThrow(token, 'token', 'String');
+        checkParamOrThrow(crawlerId, 'crawlerId', 'String');
+
+        const queryString = _.pick(options, 'token', 'nosecrets');
+
+        return requestPromise({
+            url: `${options.baseUrl}${BASE_PATH}/${userId}/crawlers/${crawlerId}`,
+            json: true,
+            method: 'GET',
+            qs: queryString,
+        });
+    },
+
+    deleteCrawler: (requestPromise, options) => {
+        const { userId, token, crawlerId } = options;
+
+        checkParamOrThrow(userId, 'userId', 'String');
+        checkParamOrThrow(token, 'token', 'String');
+        checkParamOrThrow(crawlerId, 'crawlerId', 'String');
+
+        return requestPromise({
+            url: `${options.baseUrl}${BASE_PATH}/${userId}/crawlers/${crawlerId}`,
+            json: true,
+            method: 'DELETE',
+            qs: { token },
+        });
+    },
+
+    startExecution: (requestPromise, options) => {
+        const { crawlerId, userId, token, settings } = options;
+
+        checkParamOrThrow(userId, 'userId', 'String');
+        checkParamOrThrow(token, 'token', 'String');
+        checkParamOrThrow(crawlerId, 'crawlerId', 'String');
```

Comment:
We can add validation of settings as

```
checkParamOrThrow(settings, 'settings', 'Maybe Object');
```

because for `_.isEmpty` returns `false` for `string` which is invalid.